service/key/key.go

```diff
@@ -221,3 +228,15 @@ func RemoteWriteUsernameKey() string {
 func RemoteWritePasswordKey() string {
 	return "password"
 }
+
+func IsCAPICluster(obj metav1.Object) bool {
```

Comment:
I do not understand this function, a comment would be welcome to explain what we are doing here.

From what I read I get that we only consider a cluster being CAPI when it's on v1alpha3 and it is not labeled with a giantswarm version. Is this correct?

Could we maybe only check the suffix part `giantswarm.io/version` ? So we are provider agnostic here.

Reply:
I will comment but we only check for Azure labels because only Azure is using v1alpha3 clusters. They set this annotation on the ones the Azure Operator is tracking